java/vortex-spark/src/main/java/dev/vortex/spark/read/VortexArrowColumnVector.java

```diff
@@ -0,0 +1,630 @@
+/**
+ * (c) Copyright 2025 SpiralDB Inc. All rights reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.vortex.spark.read;
+
+import com.jakewharton.nopen.annotation.Open;
+import dev.vortex.relocated.org.apache.arrow.vector.*;
+import dev.vortex.relocated.org.apache.arrow.vector.complex.ListVector;
+import dev.vortex.relocated.org.apache.arrow.vector.complex.MapVector;
+import dev.vortex.relocated.org.apache.arrow.vector.complex.StructVector;
+import dev.vortex.relocated.org.apache.arrow.vector.holders.NullableLargeVarCharHolder;
+import dev.vortex.relocated.org.apache.arrow.vector.holders.NullableVarCharHolder;
+import dev.vortex.spark.ArrowUtils;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.Decimal;
+import org.apache.spark.sql.vectorized.ColumnVector;
+import org.apache.spark.sql.vectorized.ColumnarArray;
+import org.apache.spark.sql.vectorized.ColumnarMap;
+import org.apache.spark.unsafe.types.UTF8String;
+
+@Open
+public class VortexArrowColumnVector extends ColumnVector {
```

Comment:
This is largely copied from https://github.com/apache/spark/blob/master/sql/catalyst/src/main/java/org/apache/spark/sql/vectorized/ArrowColumnVector.java but again, tweaked to use the shaded types